phoenix-core-server/src/main/java/org/apache/phoenix/coprocessor/PhoenixRegionServerEndpoint.java

```diff
@@ -133,4 +151,24 @@ public ServerMetadataCache getServerMetadataCache() {
     return ServerMetadataCacheImpl.getInstance(conf);
   }
 
+  public static TaskRunner getUncoveredIndexThreadPool(RegionCoprocessorEnvironment env) {
+    if (uncoveredIndexThreadPool == null) {
+      synchronized (PhoenixRegionServerEndpoint.class) {
```

Comment:
Why do we need to start the pool lazily? Why not start it upfront in init() specifically when we can set lower core pool size than max pool size. Core pool size can be 10 and max pool size 512 (default). Also, setting idle thread timeout will help terminate threads if they stay idle (i.e. no one uses uncovered index)